sdk/ml/azure-ai-ml/tests/model/unittests/test_model_operations.py

```diff
@@ -311,3 +311,9 @@ def test_promote_model_from_workspace(
             )
             mock_model_operation_reg.create_or_update(model_to_promote)
             mock_model_operation_reg._service_client.resource_management_asset_reference.begin_import_method.assert_called_once()
+
+    def test_model_entity_class_exist(self):
+        try:
+            from azure.ai.ml.entities import WorkspaceModelReference
```

Comment:
This test should go a bit further - can we also test passing it into 'mlclient.models.create_or_update`? This will ensure we don't break anyone who was using this as documented (even though it was incorrect)

Reply:
I think this is a good thought, my only concern is, anyway that path we will be retiring, so should it be worth keeping the test alive, the goal here is to make sure WorkspaceModelReference name stays alive, and there will be more functional tests on WorkspaceAssetlReference going forward. The other option would be when we write tests with WorkspaceAssetlReference , we can try to cast it to WorkspaceModellReference , which will be similar to this import. But this test case should not be blocking though.

Reply:
Yes if we already have tests for passing in WorkspaceAssetReference into this API, then that should be sufficient for now as they are the same model.
We can cross the bridge of breaking changes to the internal object when that time comes :)